### ISSUE TITLE
Get Chassis Status command

### ISSUE THREAD

**Issue by FerencFarkasPhD** [kontron/python-ipmi#56]
Hi,

In 2016 an issue was opened #13  for this which you closed with the comment: 

> It is fixed by adding "optional".

I do not understand where is that 'optional' field defined?
Here is what I have tried with our server:

Called the low level method:
`chasstatus=ipmi.send_message_with_name('GetChassisStatus')`
And with:
`chasstatus.__dict__`
get the result:
`{'completion_code': 0,
 'current_power_state': <pyipmi.msgs.message.Bitfield.BitWrapper at 0x22af7cb5c18>,
 'data': '',
 'front_panel_button_capabilities': 112,
 'last_power_event': <pyipmi.msgs.message.Bitfield.BitWrapper at 0x22af7cb5c88>,
 'lun': 0,
 'misc_chassis_state': <pyipmi.msgs.message.Bitfield.BitWrapper at 0x22af7cb57f0>}`

If I am calling the high level method:
`chassis_status=ipmi.get_chassis_status()`
then for:
`chassis_status.__dict__`
get this:
`{'control_fault': False,
 'fault': False,
 'interlock': False,
 'overload': False,
 'power_on': True,
 'restore_policy': 2}`

There is no 'optional' attribute. If I ask for the optional fields:
`'{0:08b}'.format(int(chasstatus.misc_chassis_state))`
then got this:
`'01000000'`

which indicates chassis identify cmd and info supported by the server.

I looked through your code in Chassys.py but I could not find this optional attribute.

Regards,
Ferenc

**Comment by hthiery** [comment]
I think first you have to understand the implementation. In the folder https://github.com/kontron/python-ipmi/tree/master/pyipmi/msgs the IPMI messages are defined.

When using 'ipmi.send_message_with_name('GetChassisStatus')' the corresponding message request object instance is created and send. The message response is also defined there and automatically created. Then in https://github.com/kontron/python-ipmi/blob/master/pyipmi/chassis.py the ChassisStatus instance is build with this information. There it seems the 'misc_chassis_state' is wrongly not added to the ChassisStatus.

**Comment by FerencFarkasPhD** [comment]
I realized that I wrongly referred to the old issue, because this is a new one related to IPMI 2.0 feature.
I made code change (and pull request) addressing both issues:
My problem is solved by adding the following two attributes for the ChassisStatus class and redefining the bitfields in GetChassisStatusRsp.
id_cmd_state_info_support, chassis_id_state

The old issue was solved by adding attribute front_panel_button_capabilities to ChassisStatus class.
If you agree with my changes than we can close this issue.
